### PR TITLE
Added multiprocessing support

### LINKED ISSUE #12: Implement multithreading to reduce execution time
**Is your feature request related to a problem? Please describe.**
Currently the execution of the script takes a lot of time. On circle-ci it takes up to 5 minutes for execution.

**Describe the solution you'd like**
The application can be multi-threaded to reduce the execution time.

### PR DESCRIPTION
**Pull request by bhumijgupta**
Fixes #12 

**Description**
The script can now use multithreading. The maximum number of threads are set to 5. Due to this, on local testing the execution time reduced from over 5 minutes to 3:20 .

**Other information**
Threading is only implemented for checking links within the license. The threads still suffer from GIL(Global Interpretor Lock) of python.

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: link_checker.py [hunk 1]
```diff
@@ -198,6 +201,37 @@ def output_summary(num_errors):
             output_write(url)
 
 
+def check_link(link, licens_name, base_url):
```
Comment (aldenstpage) [inline link_checker.py]: `licens_name` should become `license_name`

File: link_checker.py [hunk 2]
```diff
@@ -198,6 +201,37 @@ def output_summary(num_errors):
...
+    except KeyError:
+        # if there exists an <a> tag without href
+        verbose_print("Found anchor tag without href -\t", link)
+        return
+    if href[0] == "#":
+        verbose_print("Skipping internal link -\t", link)
+        return
+    status = check_existing(link)
+    with lock:
+        if status not in [200, "ignore"]:
```
[the remark this answers is not in the document]
Comment (bhumijgupta) [reply link_checker.py]: I'll add `HTTP 300 `to accepted list

File: link_checker.py (hunk 1 above)
Comment (bhumijgupta) [reply link_checker.py]: Fixed in https://github.com/creativecommons/cc-link-checker/pull/13/commits/0f48192515e4cc6554d37184b26401aaaeb9b8d7

**Comment by bhumijgupta** [comment]
After implementing `grequests`, the execution time for the script got reduced to 2:50 on Circle-ci.

**Comment by bhumijgupta** [comment]
A good option would be to `squash and merge` since the multithreading commit are not needed. This would also help to keep the commit history clean.

**Pull request merged**